scapy/layers/inet6.py

```diff
@@ -2174,18 +2181,18 @@ def names2dnsrepr(x):
     !!!  At the moment, compression is not implemented  !!!
     """
 
-    if isinstance(x, str):
-        if x and x[-1] == '\x00': # stupid heuristic
+    if isinstance(x, bytes):
+        if x and x[-1:] == b'\x00': # stupid heuristic
             return x.encode("utf8")
```

Comment:
This will fail: encode is applied on a byte string

Reply:
Totally! Fixed.